src/js/actions/layers.js

```diff
@@ -1694,10 +1684,13 @@ define(function (require, exports) {
 
         var numberRef = (typeof target === "number"),
             targetRef = numberRef ? layerLib.referenceBy.index(target) : layerLib.referenceBy[target],
-            reorderObj = layerLib.reorder(layerRef, targetRef),
-            reorderPromise = descriptor.playObject(reorderObj);
+            reorderObj = layerLib.reorder(layerRef, targetRef);
       
-        return reorderPromise.bind(this)
+        return this.transfer(historyActions.newHistoryState, document.id, strings.ACTIONS.REORDER_LAYERS)
```

Comment:
No, this makes sense. There may some structural opportunities for additional concurrency, but your explanation of why `newHistoryState` should precede everything seems sound. This confirms my understanding. It doesn't immediately seem worthwhile to have a looser kind of history event which can come before or after Photoshop's event, in the sense that it probably wouldn't buy us much, and it would make it harder for us to convince ourselves of its correctness.